doc/fundamentals/getting_data_into_pachyderm.md

```diff
@@ -1,119 +1,92 @@
+# to put <file> at <path> in the <repo>
+$ pachctl put-file <repo> <commit-id> <path> -f <file>
 
-Start and finish the commit while adding a file using ``-c``:
+# then finish the commit
+$ pachctl finish-commit <repo> <commit-id>
+```
 
-.. code-block:: shell
+Start and finish a commit while adding a file using `-c`:
 
- $ pachctl put-file -c <repo> <branch> -f <file> 
+```sh
+$ pachctl put-file <repo> <branch> <path> -c -f <file> 
+```
 
 Put data from a URL:
 
-.. code-block:: shell
-
- $ pachctl put-file <repo> <branch> -f http://url_path
-
-Add multiple files at once by using the ``-i`` option. The target file should be a list of files, paths, or URLs that you want to input all at once:
-
-.. code-block:: shell
-
- $ pachctl put-file <repo> <branch> -i <file>
-
-Pipe data from stdin into a file:
+```sh
+$ pachctl put-file <repo> <branch> <path> -c -f http://url_path
+```
 
-.. code-block:: shell
+Put data directly from an object store:
 
- $ echo "data" | pachctl put-file <repo> <branch> <path>
+```sh
+# here you can use s3://, gcs://, or as://
+$ pachctl put-file <repo> <branch> <path> -c -f s3://object_store_url
+``
 
-Add an entire directory by using the recursive flag, ``-r``:
+Add multiple files at once by using the `-i` option. The target file should be a list of files, paths, or URLs that you want to input all at once:
 
-.. code-block:: shell
+```sh
+$ pachctl put-file <repo> <branch> -c -i <file containing list of files, paths, or URLs>
+```
 
- $ pachctl put-file -r <repo> <branch> -f <dir>
+Pipe data from stdin into a data repository:
 
+```sh
+$ echo "data" | pachctl put-file <repo> <branch> <path> -c
+```
 
+Add an entire directory by using the recursive flag, `-r`:
 
-Golang Client
--------------
+```sh
+$ pachctl put-file <repo> <branch> -c -r <dir>
+```
 
-.. _autogenerated godocs: https://godoc.org/github.com/pachyderm/pachyderm/src/client#APIClient.PutFile
+### Pachyderm language clients
 
-For any Go users, we've built a Golang client so you can easily script Pachyderm commands. Check out the `autogenerated godocs`_ on ``put-file``. 
+- Go: We have a complete Golang client that will let you easily integrate pushing data to Pachyderm into your Go programs.  Check out the [godocs for put-file](https://godoc.org/github.com/pachyderm/pachyderm/src/client#APIClient.PutFile).
 
-Other Language Clients
-----------------------
-.. _protocol buffer API: https://github.com/pachyderm/pachyderm/blob/master/src/client/pfs/pfs.proto
+- Python: Our great community of users have created a [Python client for Pachyderm](https://github.com/kalugny/pypachy).  See the [included instructions](https://github.com/kalugny/pypachy#instructions) to see how you can put data into Pachyderm from Python.
 
-.. _other languages: https://developers.google.com/protocol-buffers/
+- Scala: Our users are currently working on a Scala client for Pachyderm.  Please contact us if you are interested in helping with this or testing it out.
```

Comment:
Yeah, thanks.  I couldn't find the Scala client.  Is it even public or just WIP.  I can mention that they are being updated for 1.4 and create the issue (for Python at least).